src/library/components/signup/login_modal.js

```diff
@@ -67,83 +55,87 @@ var LoginModal = function () {
       PortalPages.renderForgotPasswordModal({ oauthProviders: this.props.oauthProviders })
     },
 
+    handleRegister: function (e) {
+      e.preventDefault()
+      PortalPages.renderSignupModal({ oauthProviders: this.props.oauthProviders })
+    },
+
     render: function () {
       // console.log("INFO rendering LoginModal with props", this.props);
 
-      var providerComponents = []
+      let providerComponents = []
 
       if (enableAuthProviders && this.props.oauthProviders) {
-        var providers = this.props.oauthProviders
+        let providers = this.props.oauthProviders
         providers.sort(function (a, b) { return (a.name > b.name) ? 1 : ((b.name > a.name) ? -1 : 0) }) // sort providers alphabetically by name
-        for (var i = 0; i < providers.length; i++) {
+        for (let i = 0; i < providers.length; i++) {
+          if (this.props.afterSigninPath) {
+            providers[i].directPath += '?after_sign_in_path=' + encodeURIComponent(this.props.afterSigninPath)
```

Comment:
Looking at this again, it seems like this could be a problem. Each time render is called it will be modifying direct path of the providers. So I'd expect that if you open the login_modal and then close it and open it again you'll get the after_sign_in_path added twice.
So a better approach would be to assign the directPath to a local variable:
```
let directPath = providers[i].directPath
```
Then modify that local variable if there is an afterSigninPath.
Then use that local variable on line 76.

Reply:
@scytacki Thanks for catching that. I've made the change.